src/components/utils/src/socket_posix.cc

```diff
@@ -36,191 +36,262 @@
 
 #include "utils/socket.h"
 
-namespace utils {
+#include <algorithm>
 
-class Socket::Impl {
- public:
-  Impl();
-  explicit Impl(int socket);
-  ~Impl();
+#include "utils/winhdr.h"
+#include "utils/socket.h"
 
-  Impl(Impl& rh);
-  Impl& operator=(Impl& rh);
+CREATE_LOGGERPTR_GLOBAL(logger_ptr, "Utils.TcpSocket")
+
+namespace {
+
+  bool CloseSocket(int& socket) {
+    LOG4CXX_AUTO_TRACE(logger_ptr);
+    if (NULL == socket) {
+      LOG4CXX_DEBUG(logger_ptr, "Socket " << socket << " is not valid. Skip closing.");
```

Comment:
Usage of the `CloseSocket` with 0 is the positive case. E.g. when we return new TcpSocketConnection out of the accept. Thus we shouldn't use high log level. This is really debug.